runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -183,8 +184,8 @@ public void write(@Nonnull IToken token, @Nonnull Object data, @Nonnull CacheOpt
                     // Large writes are also rejected right away.
                     throw ex;
                 }
-            } catch (WriteSizeException we) {
-                throw we;
+            } catch (WriteSizeException | QuotaExceededException ie) {
+                throw ie;
```

Comment:
Sleep here?

Reply:
Interesting suggestion I was thinking of sleeping in the retry logic up in the caller, but I think we can add a sleep here too.